installer/lib/mix/tasks/phx.new.ex

```diff
@@ -145,7 +145,7 @@ defmodule Mix.Tasks.Phx.New do
       webpack_pending = install_webpack(install?, project)
       Task.await(compile, :infinity)
 
-      if Project.webpack?(project) and !System.find_executable("npm") do
+      if Project.webpack?(project) and !elem(node_package_manager(), 0) do
```

Comment:
Could we avoid using `elem` here? The pattern matching approach used in `install_webpack` is much clearer.

Another option is to return the package manager command or `nil` from `node_package_manager()`

```
if Project.webpack?(project) and !node_package_manager() do

defp node_package_manager() do
  cond do
    System.find_executable("yarn") -> 
      "yarn"
    System.find_executable("npm") ->
      "npm install"
    true ->
      nil
  end
end
```

Reply:
I really like the explicit return of "npm install" for the command logged into the console if npm/yarn is not available, but maybe it's enough to let `install_webpack` default to that command.

Reply:
@LostKobrakai that's a good point. In that case, I'd go for:

```
   {pkm_available?, _} = node_package_manager()
    if Project.webpack?(project) and !pkm_available? do
```